controller/watch/watch_test.go

```diff
@@ -14,12 +14,125 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package watch_test
+package watch
 
 import (
+	"reflect"
 	"testing"
+	"time"
+
+	"k8s.io/client-go/1.5/pkg/api/unversioned"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	extv1beta1 "k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/1.5/pkg/watch"
 )
 
+func TestDeploymentWatcher(t *testing.T) {
+	const (
+		evtChTimeout = 100 * time.Millisecond
+	)
+	fakeWatcher := watch.NewFake()
+	deplIface := &fakeDeploymentInterface{
+		listRet: &extv1beta1.DeploymentList{
+			TypeMeta: unversioned.TypeMeta{},
+			ListMeta: unversioned.ListMeta{},
+			Items: []extv1beta1.Deployment{
+				extv1beta1.Deployment{
```

Comment:
oh, btw, running `make format` at the repo root will get rid of them -- the auto-simplify code feature of Go will kick in.